virttest/utils_misc.py

```diff
@@ -2564,12 +2564,18 @@ def __get_unit_index(M):
         raise ValueError("Invalid data size format 'value_str=%s'" % value_str)
     from_index = __get_unit_index(unit)
     to_index = __get_unit_index(order_magnitude)
-    scale = int(factor) ** (to_index - from_index)
-    if scale > 0:
-        data_size = float(value) / abs(scale)
+    scale = int(factor) ** abs(to_index - from_index)
+    if to_index == from_index:
+        data_size = float(value)
```

Comment:
Yes, your are right, seems I didn't catch the whole meaning here. I will update this patch.